lib/srv/reexec.go

```diff
@@ -93,6 +94,30 @@ type ExecCommand struct {
 
 	// IsTestStub is used by tests to mock the shell.
 	IsTestStub bool `json:"is_test_stub"`
+
+	// RemoteAddr is the address of the remote host.
+	RemoteAddr string `json:"remote_addr"`
+
+	// The hostname of the node.
+	Hostname string `json:"hostname"`
+}
+
+func createUaccSession(ttyName string, c *ExecCommand) error {
+	remoteStringIP, _, _ := net.SplitHostPort(c.RemoteAddr)
+	remoteIP := net.ParseIP(remoteStringIP)
+	err := uacc.Open(c.Login, c.Hostname, remoteIP, ttyName)
```

Comment:
Technically it can but that will never happen since the IP is always well formed unless something corrupt the JSON in flight. Though removing the wrappers may be a good idea. Not sure though.

Reply:
It doesn't look like you're using `RemoteAddr` directly anywhere. Why not parse it the parent and send something you can directly use? It simplifies your implementation no more need for wrappers.

A pattern I tried to maintain was do everything that you possibly can in the parent and then just send a bundle to the child to execute without thinking since we don't have much opportunity for logging in the child.

Reply:
:+1: done